xbmc/addons/interfaces/Filesystem.cpp

```diff
@@ -503,6 +504,40 @@ int Interface_Filesystem::get_file_chunk_size(void* kodiBase, void* file)
   return static_cast<CFile*>(file)->GetChunkSize();
 }
 
+const char* Interface_Filesystem::get_property(void* kodiBase, void* file, int type, const char *name)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (addon == nullptr || file == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_Filesystem::%s - invalid data (addon='%p', file='%p')", __FUNCTION__, addon, file);
+    return "";
+  }
+
+  XFILE::FileProperty internalType;
+  switch (type)
+  {
+  case ADDON_FILE_PROPERTY_RESPONSE_PROTOCOL:
+    internalType = XFILE::FILE_PROPERTY_RESPONSE_PROTOCOL;
+    break;
+  case ADDON_FILE_PROPERTY_RESPONSE_HEADER:
+    internalType = XFILE::FILE_PROPERTY_RESPONSE_HEADER;
+    break;
+  case ADDON_FILE_PROPERTY_CONTENT_TYPE:
+    internalType = XFILE::FILE_PROPERTY_CONTENT_TYPE;
+    break;
+  case ADDON_FILE_PROPERTY_CONTENT_CHARSET:
+    internalType = XFILE::FILE_PROPERTY_CONTENT_CHARSET;
+    break;
+  case ADDON_FILE_PROPERTY_MIME_TYPE:
+    internalType = XFILE::FILE_PROPERTY_MIME_TYPE;
+    break;
+  default:
+    CLog::Log(LOGERROR, "Interface_Filesystem::%s - invalid data (addon='%p', file='%p')", __FUNCTION__, addon, file);
+    return "";
+  };
+  return static_cast<CFile*>(file)->GetProperty(internalType, name).c_str();
```

Comment:
This will most probably crash if a std::string for param 2 gets constructed and `name` is `nullptr`.